### PR TITLE
add TitleCase to enum names to support iOS enum conventions

### PR DESCRIPTION
**Pull request by giannic**
Hello @nicks, @jyhsu, 

Please review the following commits I made in branch 'giannic-ios-enum'.

07b8905d4e492e012e520658de1257b96b1ea3cf (2015-07-17 11:43:49 -0700)
add TitleCase to enum names to support iOS enum conventions

R=@nicks
R=@jyhsu

Test plan: 'manual'

### CODE REVIEW

File: lib/helper.js [hunk 1]
```diff
@@ -57,7 +57,8 @@ exports.extend = function (var_args) {
  * @return {string}
  */
 exports.toCamelCase = function(str) {
-  return String(str).replace(/\_([a-z])/g, function(all, match) {
+  var lowerCase = str.toLowerCase()
```
Comment (nicks) [inline lib/helper.js]: this doesn't seem right to me. it also needs a unit test

**Comment by nicks** [comment]
(note that the unit tests are broken by this change)

**Comment by giannic** [comment]
updated test and added a field that shows the change more clearly!

**Comment by nicks** [comment]
lgtm

**Pull request merged**